### PR TITLE
EVSS: WIP rename project

### PR DESCRIPTION
**Pull request by carlmjohnson**
How should we handle the URL change? Do a redirect?

LMK how the rest looks. It's a big change set, so I don't want anything to slip through the cracks.

Refs https://github.com/department-of-veterans-affairs/sunsets-team/issues/287

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ before_script:
 - export VBA_MAPSERVER_URL='https://services3.arcgis.com/aqgBd3l68G8hEFFE/ArcGIS/rest/services/VBA_Facilities/FeatureServer/0'
 - export MOCK_MVI_SERVICE=false
 - export GOV_DELIVERY_SERVER='stage-tms.govdelivery.com'
+- export ES_URL='https://test.vets.gov'
+- export ES_CLIENT_CERT_PATH='/fake/client/cert/path'
+- export ES_CLIENT_KEY_PATH='/fake/client/key/path'
+
```
Comment (carlmjohnson) [inline .travis.yml]: I had to merge master to make tests pass but now they're showing up here. Maybe do a separate PR to put master and f/all-claims into sync?

File: config/routes.rb [hunk 2]
```diff
@@ -17,8 +17,13 @@
 
     resource :education_benefits_claims, only: [:create]
 
+    resource :health_care_application, only: [:create] do
+      resource :healthcheck, only: [:show], to: 'health_care_applications#healthcheck'
+    end
+
     resource :disability_rating, only: [:show]
-    resources :disability_claims, only: [:index, :show] do
+    # TODO: How to handle old URLs? Redirect?
```
Comment (austin-martinez) [inline config/routes.rb]: The following will point both routes to the same controller, and then we can remove the old routes once the FE is updated
```ruby
# TODO: Remove this resource/subresource when FE is updated
resources :disability_claims, only: [:index, :show], controller: 'evss_claims', as: :evss_claim do
      post :request_decision, on: :member
      resources :documents, only: [:create]
    end
```

File: app/serializers/user_serializer.rb [hunk 3]
```diff
@@ -31,7 +31,7 @@ def services
       BackendServices::EDUCATION_BENEFITS
     ]
     service_list += [BackendServices::RX, BackendServices::MESSAGING] if object.can_access_mhv?
-    service_list << BackendServices::DISABILITY_BENEFITS if object.can_access_evss?
+    service_list << BackendServices::EVSS_BENEFITS if object.can_access_evss?
```
Comment (austin-martinez) [inline app/serializers/user_serializer.rb]: This will break the frontend immediately once its merged. Could we add the new service name as well as the old one to the list? Also, not sure what the best service name is here, maybe `claim-status`?

File: app/workers/evss/document_upload.rb [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
-class DisabilityClaim::DocumentUpload
+class EVSSClaim::DocumentUpload
```
Comment (austin-martinez) [inline app/workers/evss/document_upload.rb]: This change may impact any queued jobs when its deployed as the class name stored will be the old name. Not sure best way to handle it, a migration/task that removes it from the queue and queues it with this class? The number of jobs should be small or even zero, but still a risk. Another option would be to keep class with old name and pass through arguments to this class, or have it inherit from this class

File: app/uploaders/evss_claim_document_uploader.rb [hunk 5]
```diff
@@ -13,7 +13,7 @@ def initialize(user_uuid, tracked_item_id)
   end
 
   def store_dir
-    store_dir = "disability_claim_documents/#{@user_uuid}"
+    store_dir = "evss_claim_documents/#{@user_uuid}"
```
Comment (austin-martinez) [inline app/uploaders/evss_claim_document_uploader.rb]: This could cause an issue with any queued jobs that have documents uploaded to the old `store_dir` and then try to access them from the new `store_dir`

File: app/serializers/user_serializer.rb (hunk 3 above)
Comment (carlmjohnson) [inline app/serializers/user_serializer.rb]: I think we should rename the constant but leave its value the same as before, then update the front end to use either the old or new value then update the backend to send a new value.

File: app/uploaders/evss_claim_document_uploader.rb (hunk 5 above)
Comment (carlmjohnson) [inline app/uploaders/evss_claim_document_uploader.rb]: I think the solution for queued tasks is to have all new jobs use new classes but leave the old classes with old upload location until all the old jobs are processed and remove them in a separate production push.

File: app/serializers/user_serializer.rb (hunk 3 above)
Comment (austin-martinez) [inline app/serializers/user_serializer.rb]: That would require 4 updates vs 3

1. Update here to use different constant name
2. Update front end to use either value
3. Update backend to send new value
4. Update front end to only use new value

vs.

1. Update here to send 2 values
2. Update frontend to use new value
3. Update backend to only send new value
